src/XrdCms/XrdCmsCluster.cc

```diff
@@ -1800,7 +1807,17 @@ XrdCmsNode *XrdCmsCluster::SelbyLoad(SMask_t mask, XrdCmsSelector &selR)
                                   || (reqSS && np->isNoStage)))
               {selR.xFull = true; continue;}
            if (!sp) sp = np;
-              else{if (selR.needSpace)
+              else{
+                 if (Config.P_randlb==1){
+                   //add 1 to the inverse load, this is to allow some selection in case reported loads hit 100
+                   weighed[i] = selCap + static_cast<int>(101 - np->myLoad +
+                     std::pow(101 - np->myLoad,
+                     std::log(100)-std::log(std::min(std::max(Config.P_fuzz,1),100)))/2);
+                   selCap += static_cast<int>(101 - np->myLoad +
+                     std::pow(101 - np->myLoad,
+                     std::log(100)-std::log(std::min(std::max(Config.P_fuzz,1),100)))/2);
```

Comment:
You should assign `std::log(100)-std::log(std::min(std::max(Config.P_fuzz,1),100)))/2` to a variable and reuse it instead of computing this twice. You can also use [std::clamp](https://en.cppreference.com/w/cpp/algorithm/clamp) here instead of the combination of `std::min` + `std::max`, since we have moved to C++17 as our baseline C++ standard on all platforms. Since this expression also doesn't depend on the loop variable, it should be hoisted outside of the `for` loop to be computed only once. But also, is there no way to have this done without `log` and `pow`? These are expensive things to be called frequently, and will burn lots of CPU when there is heavy load in the system. If the range is limited, you should consider using a polynomial approximation of the `log`, I think the exact shape here is not very important, just the general behavior in the range 1 to 100.

Reply:
I've moved the calculation outside the loop and used std::clamp. Exponentials are needed to show significant differences in the load distribution according to the specified fuzz. I've replaced the log function with a scaled percentile, it was used to prevent the exponentials from triggering an integer overflow.